src/sentry/runner/commands/createdetector.py

```diff
@@ -0,0 +1,130 @@
+from re import sub
+
+import click
+
+from sentry.runner.decorators import configuration
+
+DETECTORS_DIR = "src/sentry/performance_issues/detectors"
+
+
+def snake_to_upper_camel_case(value: str) -> str:
+    """
+    Converts a string from snake_case to UpperCamelCase
+    """
+    return "".join(word.capitalize() for word in value.strip("_").split("_"))
+
+
+@click.command()
+@click.argument("name", required=True)
+@configuration
+def createdetector(name: str) -> None:
+    """Create a new Detector. Name passed must be in snake_case.
+    Example invocation: sentry createdetector solar_eclipse would create a new
+    SolarEclipseDetector and integrate it into the Detector-tracking files."""
+    detector_filename = f"{DETECTORS_DIR}/{name}_detector.py"
+    detector_classname = snake_to_upper_camel_case(name) + "Detector"
+    group_type_classname = snake_to_upper_camel_case(name) + "GroupType"
+    detector_type = name.upper()
+    option_name = f"performance.issues.{name}.problem-creation"
+
+    detector_file_contents = f"""from sentry.models.organization import Organization
+from sentry.models.project import Project
+
+from ..base import DetectorType, PerformanceDetector
+from ..types import Span
+
+
+class {detector_classname}(PerformanceDetector):
+
+    type = DetectorType.{detector_type}
+    settings_key = DetectorType.{detector_type}
+
+    def is_creation_allowed_for_organization(self, organization: Organization | None) -> bool:
+        return False  # TODO
+
+    def is_creation_allowed_for_project(self, project: Project) -> bool:
+        return False  # TODO
+
+    def visit_span(self, span: Span) -> None:
+        pass  # TODO
+
+    def on_complete(self) -> None:
```

Comment:
i think only a few detectors use `on_complete` so it _might_ be confusing to include it